features/netsocket/emac-drivers/TARGET_Cypress/TARGET_PSOC6/TARGET_WHD/interface/whd_emac.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "cmsis_os.h"
+#include "whd_emac.h"
+#include "lwip/etharp.h"
+#include "lwip/ethip6.h"
+#include "mbed_shared_queues.h"
+#include "whd_wlioctl.h"
+#include "whd_buffer.h"
+#include "whd_buffer_api.h"
+#include "cybsp_api_wifi.h"
+
+WHD_EMAC::WHD_EMAC(whd_interface_role_t role)
+    : interface_type(role)
+{
+}
+
+WHD_EMAC::WHD_EMAC()
+    : interface_type(WHD_STA_ROLE)
+{
+}
+
+WHD_EMAC &WHD_EMAC::get_instance() {
+    return get_instance(WHD_STA_ROLE);
+}
+
+WHD_EMAC &WHD_EMAC::get_instance(whd_interface_role_t role) {
+    static WHD_EMAC emac_sta(WHD_STA_ROLE);
+    static WHD_EMAC emac_ap(WHD_AP_ROLE);
+    return role == WHD_AP_ROLE ? emac_ap : emac_sta;
+}
+
+uint32_t WHD_EMAC::get_mtu_size() const
+{
+    return WHD_PAYLOAD_MTU;
+}
+
+uint32_t WHD_EMAC::get_align_preference() const
+{
+    return 0;
+}
+
+void WHD_EMAC::add_multicast_group(const uint8_t *addr)
+{
+    memcpy(multicast_addr.octet, addr, sizeof(multicast_addr.octet));
+    whd_wifi_register_multicast_address(ifp, &multicast_addr);
+}
+
+void WHD_EMAC::remove_multicast_group(const uint8_t *addr)
+{
+    whd_wifi_unregister_multicast_address(ifp, &multicast_addr);
+}
+
+void WHD_EMAC::set_all_multicast(bool all)
+{
+    //TODO: new API call for this in the whd interface to be added.
+    //whd_wifi_set_iovar_value(ifp, IOVAR_STR_ALLMULTI, all);
+}
+
+void WHD_EMAC::power_down()
+{
+    powered_up = false;
+    whd_wifi_off(ifp);
+    whd_deinit(ifp);
+}
+
+bool WHD_EMAC::power_up()
+{
+	drvp = *(cybsp_get_wifi_driver());
```

Comment:
the code looks unaligned in this function (these 2 lines)

Reply:
Fixed by applying astyle in https://github.com/ARMmbed/mbed-os/pull/10692/commits/c115f23028c674329e859c26f4b7ad06f5d3d66c#diff-38f33b303ec6a94c6f64f1bd8778ca90R91